infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -222,6 +228,12 @@ public void sendWithBuffering(List<LogReplicationEntryMsg> dataToSend, String me
     public LogReplicationEntryMsg resend() {
```

Comment:
This method gets invoked on Snapshot sync also.  For snapshot sync, the timeout reason will most likely be different.  We will have to decide if the backoff must be introduced there as well.